server/index.js

```diff
@@ -24,14 +24,14 @@ const internalPrefixes = [
 ]
 
 export default class Server {
-  constructor ({ dir = '.', dev = false, staticMarkup = false, quiet = false } = {}) {
+  constructor ({ dir = '.', dev = false, staticMarkup = false, quiet = false, conf = {} } = {}) {
     this.dir = resolve(dir)
     this.dev = dev
     this.quiet = quiet
     this.router = new Router()
-    this.hotReloader = dev ? new HotReloader(this.dir, { quiet }) : null
+    this.hotReloader = dev ? new HotReloader(this.dir, { quiet, conf }) : null
     this.http = null
-    this.config = getConfig(this.dir)
+    this.config = getConfig(this.dir, conf)
```

Comment:
If we've a config via the option, it shouldn't read from the filesystem. Change the code to reflect that.

And make sure to get default configurations. May be we need to add method called `withDefaults` to `server/config.js`